puzzle.js

```diff
@@ -218,11 +228,11 @@ function PuzzleEntry(p, index) {
     modes.reverse();
 
     this.options = {};
-    modes.forEach(m => { for (const [key, value] of Object.entries(puzzleModes[m])) { this.options[key] = value; } });
+    modes.forEach(m => { for (const [key, value] of Object.entries(puzzleModes[m])) { this.options[key] = parseAsJsObject(value); } });
```

Comment:
Thinking on this, I bet it breaks Sudoku puzzles when allowed characters are turned into an integer...

Reply:
I get what it's trying to fix but I think it should only be used for a whitelist of properties as a final options processing step.

Reply:
Thanks for fixing the parser!